### PR TITLE
fix: avoid useless space before the closing of a property-less tag

### PR DESCRIPTION
**Pull request by mdealencar**
Method `Element.as_str()` always adds a space before the closing `>` of a tag that has no properties. This PR fixes that.

### CODE REVIEW

**Comment by orsinium** [comment]
Suggestion: this should be simpler and achieve the same goal:

```python
        props = " ".join(f'{k}="{v}"' for k, v in self.as_dict().items())
        if self.data:
            if props:
                props += " "
            props += " ".join(f'data-{k}="{v}"' for k, v in self.data.items())
```

**Comment by mdealencar** [comment]
I think your suggestion does not solve the issue, as the method is returning a hard-coded space after element_name: `f"<{self.element_name} {props}>"`

```
    def as_str(self) -> str:
        props = " ".join(f'{k}="{v}"' for k, v in self.as_dict().items())
        if self.data:
            props += " " + " ".join(f'data-{k}="{v}"' for k, v in self.data.items())
        if self.text:
            return f"<{self.element_name} {props}>{self.text}</{self.element_name}>"
        if self.elements:
            content = "".join(self._as_str(e) for e in self.elements)
            return f"<{self.element_name} {props}>{content}</{self.element_name}>"
        return f"<{self.element_name} {props}/>"
```

**Comment by jjjnovak** [comment]
I resolve this I think for self.text in my just new diffs .... - it could be moved above the ifs I suppose ...

**Comment by orsinium** [comment]
Hey, thank you for your contribution. I like the idea and I appreciate the time you've spent on this. However, I find the implementation to be hard to follow and I struggle to properly review it yet alone to maintain. The PR has been in a limbo for a while, and I think at this point it's better to close it. If you come up with an easier solution without , maybe solving only some of the cases, I'd be happy to review that.

If there are any issues with the svg.py output formatting, you can always post-process it using a code formatter like [prettier](https://prettier.io/):

```bash
python3 ./examples/animate.py | prettier --parser html
```
